File: basic/java/1059. C语言竞赛(20).java
import java.io.BufferedReader;
import java.io.InputStreamReader;
import java.util.ArrayList;
import java.util.Arrays;
import java.util.HashMap;
import java.util.HashSet;
import java.util.List;
import java.util.Map;
import java.util.Scanner;
import java.util.Set;

public class Main {
	public static void main(String[] args) {
		Scanner in=new Scanner(System.in);
		Map IdtoRank=new HashMap(); 
		boolean mark[]=new boolean[10000];
		int n=in.nextInt();
		for(int i=1;i<=n;++i) {
			int id=in.nextInt();
			IdtoRank.put(id, i);
		}
		int k=in.nextInt();
		for(int i=0;i<k;++i) {
			int id=in.nextInt();
			System.out.printf("%04d: ", id);
			if(IdtoRank.get(id)==null) {
				System.out.println("Are you kidding?");
				continue;
			}else if(mark[id]) {
				System.out.println("Checked");
				continue;
			}
			int rank=Integer.parseInt(IdtoRank.get(id).toString());
			if(rank==1) System.out.println("Mystery Award");
			else if(judge(rank))System.out.println("Minion");
			else System.out.println("Chocolate");
			mark[id]=true;
		}
	}
	
	public static boolean judge(int num) {
		if(num<=1)return false;
		for(int i=2;i*i<=num;++i)
			if(num%i==0)return false;
		return true;
	}
}

File: basic/1059. C语言竞赛(20).cpp
#include<iostream>
#include<map>
using namespace std;
bool judge(int num){
	if(num<=1)return false;
	for(int i=2;i*i<=num;++i)
		if(num%i==0)return false;
	return true;
}
int main(){
	int n;
	cin>>n;
	map<int,int> IdtoRank;
	map<int,bool> mark;
	for(int i=1;i<=n;++i){
		int id;cin>>id;
		IdtoRank[id]=i;
	}
	int k;
	cin>>k;
	while(k--){
		int id;
		cin>>id;
		if(mark[id]){
			printf("%04d: Checked\n",id);
			continue;
		}
		if(IdtoRank[id]==0){
			printf("%04d: Are you kidding?\n",id);
			continue;
		}
		if(IdtoRank[id]==1){
			printf("%04d: Mystery Award\n",id);
			mark[id]=true;
			continue;
		}
		if(judge(IdtoRank[id])){
			printf("%04d: Minion\n",id);
			mark[id]=true;
			continue;
		}
		printf("%04d: Chocolate\n",id);
		mark[id]=true;
		
	}
	return 0;
}
